module/tabbed.lua

```diff
@@ -22,43 +22,101 @@ local bar = require(tostring(...):match(".*bling") .. ".widget.tabbar." ..
 
 tabbed = {}
 
--- used to change focused tab relative to the currently focused one 
+local get_tab_buttons = function(c)
+    return gears.table.join(
+        awful.button(
+            {}, 1,
+            function()
+                local tabobj = c.tab_obj
+                tabbed.switch_to(tabobj, c.tab_idx)
+                c.move_timer = gears.timer({
+                    timeout = 0.3,
+                    callback = function()
+                        awful.mouse.client.move(c)
+                        c.move_timer = nil
+                    end,
+                    single_shot = true,
+                })
+                c.move_timer:start()
+            end,
+            function()
+                if (c.move_timer) then
+                    c.move_timer:stop()
+                    c.move_timer = nil
+                end
+            end
+        ),
+        awful.button({}, 2, tabbed.pop),
+        awful.button({}, 3,
+            function()
+                local tabobj = c.tab_obj
+                tabbed.switch_to(tabobj, c.tab_idx)
+                c.resize_timer = gears.timer({
+                    timeout = 0.3,
+                    callback = function()
+                        awful.mouse.client.resize(c)
+                        c.resize_timer = nil
+                    end,
+                    single_shot = true,
+                })
+                c.resize_timer:start()
+            end,
+            function()
+                if (c.resize_timer) then
+                    c.resize_timer:stop()
+                    c.resize_timer = nil
+                end
+            end
+        )
+    )
+end
+
+-- used to change focused tab relative to the currently focused one
 tabbed.iter = function(idx)
     if not idx then idx = 1 end
-    if not client.focus or not client.focus.bling_tabbed then return end
-    local tabobj = client.focus.bling_tabbed
+    if not client.focus or not client.focus.tab_obj then return end
+    local tabobj = client.focus.tab_obj
     local new_idx = (tabobj.focused_idx + idx) % #tabobj.clients
     if new_idx == 0 then new_idx = #tabobj.clients end
     tabbed.switch_to(tabobj, new_idx)
 end
 
 -- removes a given client from its tab object
 tabbed.remove = function(c)
-    if not c or not c.bling_tabbed then return end
-    local tabobj = c.bling_tabbed
-    table.remove(tabobj.clients, tabobj.focused_idx)
-    awful.titlebar.hide(c, bar.position)
```

Comment:
That line should probably stay. The tabbar currently doesn't hide at all when removing a client (if you haven't set any titlebars at that side that is).